collections/soldiers.js

```diff
@@ -44,4 +44,28 @@ function newSoldier(collection, data, errorHandler) {
   }
 }
 
-module.exports.newSoldier = newSoldier;
\ No newline at end of file
+function getSoldier(collection, name, id, errorHandler) {
+  let searchedParams = {};
+
+  if (name) {
```

Comment:
i commented on this. this code is the same as 
searchedParams["name"] = name
searchedParams["id"] = id